src/utils/daos_hdlr.c

```diff
@@ -2566,95 +2576,274 @@ fs_copy(struct file_dfs *src_file_dfs,
 		if (rc != 0) {
 			fprintf(stderr, "failed to connect to destination "
 				"pool: %d\n", rc);
 			D_GOTO(out, rc);
 		}
-		rc = daos_cont_open(fa->src_poh, fa->src_c_uuid, DAOS_COO_RW,
-				    &fa->src_coh, src_cont_info, NULL);
+		rc = daos_cont_open(ca->src_poh, ca->src_c_uuid, DAOS_COO_RW,
+				    &ca->src_coh, src_cont_info, NULL);
 		if (rc != 0) {
 			fprintf(stderr, "failed to open source "
 				"container: %d\n", rc);
 			D_GOTO(err_src_root, rc);
 		}
-		rc = dfs_mount(fa->src_poh, fa->src_coh, O_RDWR,
-			       &src_file_dfs->dfs);
-		if (rc) {
-			fprintf(stderr, "dfs mount on source failed: %d\n", rc);
-			D_GOTO(err_src_dfs, rc);
+		if (is_posix_copy) {
+			rc = dfs_mount(ca->src_poh, ca->src_coh, O_RDWR,
+				       &src_file_dfs->dfs);
+			if (rc) {
+				fprintf(stderr, "dfs mount on source "
+					"failed: %d\n", rc);
+				D_GOTO(err_src, rc);
+			}
 		}
 	}
 
-	/* open dst pool, dst cont, and mount dfs */
-	if (dst_file_dfs->type == DAOS) {
-		rc = daos_pool_connect(fa->dst_p_uuid, sysname,
-				       DAOS_PC_RW, &fa->dst_poh, NULL, NULL);
+	/* only need to query if source is not POSIX, since
+	 * this connect call is used by the filesystem and clone
+	 * tools
+	 */
+	if (src_file_dfs->type != POSIX) {
+		dpe_types[0] = ca->cont_prop_layout;
+		dpe_types[1] = ca->cont_prop_oid;
+
+		rc = dm_get_cont_prop(ca->src_coh, sysname, src_cont_info, size,
+				      dpe_types, dpe_vals);
 		if (rc != 0) {
-			fprintf(stderr, "failed to connect to destination "
-				"pool: %d\n", rc);
+			fprintf(stderr, "failed to query source "
+				"container: %d\n", rc);
+			D_GOTO(err_src, rc);
+		}
+
+		ca->cont_layout = dpe_vals[0];
+		ca->cont_oid = dpe_vals[1];
+
+		props = daos_prop_alloc(2);
+		if (props == NULL) {
+			fprintf(stderr, "Failed to allocate prop (%d)", rc);
 			D_GOTO(out, rc);
 		}
-		rc = daos_cont_open(fa->dst_poh, fa->dst_c_uuid, DAOS_COO_RW,
-				    &fa->dst_coh, dst_cont_info, NULL);
-		if (rc != 0) {
-			fprintf(stderr, "failed to open destination "
-				"container: %d\n", rc);
-			D_GOTO(err_dst_root, rc);
+		props->dpp_entries[0].dpe_type = ca->cont_prop_layout;
+		props->dpp_entries[0].dpe_val = ca->cont_layout;
+
+		/* only set max oid on created dest containers
+		 * if this is a non-posix copy
+		 */
+		if (!is_posix_copy) {
+			props->dpp_entries[1].dpe_type = ca->cont_prop_oid;
+			props->dpp_entries[1].dpe_val = ca->cont_oid;
 		}
-		rc = dfs_mount(fa->dst_poh, fa->dst_coh, O_RDWR,
-			       &dst_file_dfs->dfs);
-		if (rc != 0) {
-			fprintf(stderr, "dfs mount on destination "
-			"failed: %d\n", rc);
-			D_GOTO(err_dst_dfs, rc);
+	}
+
+	/* open dst pool, dst cont, and mount dfs */
+	if (dst_file_dfs->type == DAOS) {
+		/* only connect if destination pool wasn't already opened */
+		if (!uuid_is_null(ca->dst_p_uuid)) {
+			if (!daos_handle_is_valid(ca->dst_poh)) {
+				rc = daos_pool_connect(ca->dst_p_uuid, sysname,
+		       	       			       DAOS_PC_RW, &ca->dst_poh,
+				       		       NULL, NULL);
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs